src/main/scala/devices/gpio/GPIO.scala

```diff
@@ -172,8 +193,11 @@ abstract class GPIO(busWidthBytes: Int, c: GPIOParams)(implicit p: Parameters)
     swPinCtrl(pin).pue    := pueReg.io.q(pin)
     swPinCtrl(pin).oval   := portReg(pin)
     swPinCtrl(pin).oe     := oeReg.io.q(pin)
-    swPinCtrl(pin).ds     := dsReg(pin)
+    swPinCtrl(pin).ds     := dsReg(0)(pin)
```

Comment:
Since we're doing it this way, let's add a `require(dsWidth > 0)` to `GPIOParams`